src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2924,27 +2924,138 @@ LogicalResult ONNXSplitV11Op::inferShapes(
 //===----------------------------------------------------------------------===//
 
 LogicalResult ONNXSplitToSequenceOp::verify() {
-  ONNXSplitToSequenceOpAdaptor operandAdaptor(*this);
```

Comment:
I'm curious about the use of operandAdaptor across the code base: it seems like it's used as an unnecessary detour to access inputs and attributes in places like shape inference where we have direct access to them through the op itself

my interpretation of the MLIR docs
https://mlir.llvm.org/docs/OpDefinitions/#operand-adaptors
and the tutorial example:
https://mlir.llvm.org/docs/Tutorials/Toy/Ch-5/#conversion-patterns
is that operandAdaptor is meant for cases where only the inputs and attributes are available, but not the op, and we want to access inputs and attributes with named methods in another way than through an op

but I'm happy to reintroduce operandAdaptor here if that's what everybody else prefers

Reply:
I see how in ConversionPatterns, like in ONNXMatMulOpLowering as an example, operandAdaptor allows you to access the lowered types, but AFAICT this doesn't apply to the shape checking and inference logic in the verify() and most inferShapes() methods in ONNXOps.cpp

I can see why the shape inference logic in the ShapeHelper classes takes operandAdaptor input to the computeShape method, because this is reused (through IndexExpr) across inferShapes() in ONNXOps.cpp and in the matchAndRewrite method in the corresponding ConversionPattern (e.g. ONNXMatMulOpShapeHelper is called from both ONNXMatMulOp::inferShapes and ONNXMatMulOpLowering::matchAndRewrite)

in summary, I believe that operandAdaptor is an unnecessary detour in the case of inferShapes() logic that doesn't use IndexExpr to become reusable for conversion logic, namely in all the inferShapes() methods that are implemented fully in ONNXOps.cpp without calling any ShapeHelpers

Reply:
That make sense. I would wait for @chentong319 to chime in before taking a decision, but your conclusions feel right to me. Maybe @tungld also has an opinion.

Reply:
@sorenlassen That makes sense to me too.

Reply:
@sorenlassen @AlexandreEichenberger @tungld  I agree that there is no need to use Adaptor because shape inference does not use the pattern rewriting framework. We may clean up the code when we have time.